### DISCUSSION TITLE
Q2

### DISCUSSION THREAD

**Discussion by Madura-Prasad** [Madura-Prasad/Bumble-Bee-Loan-System#3] [Q&A] [answered]
What are the available scripts and their purposes in the Create React App-based front-end project?

**Comment by SithumDilshan28** [comment, marked as answer]
npm start: This script runs the React app in development mode. It starts a local development server and opens the app in your browser at http://localhost:3000. The page will automatically reload when you make changes to the code.
npm test: This script launches the test runner in the interactive watch mode. It allows you to run tests and see the results in real-time as you make changes to your test files.
npm run build: This script builds the app for production. It creates a optimized and minified version of your app in the build folder. This build is ready to be deployed to a web server for production use.
npm run eject: This script is a one-way operation that should be used with caution. It "ejects" your Create React App project, giving you full control over the configuration and build tools. Once ejected, you cannot go back. Ejecting is usually done when you need to customize the build configuration beyond the capabilities provided by Create React App.
